components/timezone/src/time_zone.rs

```diff
@@ -21,123 +20,94 @@ use tinystr::TinyAsciiStr;
-            metazone_id: Some(MetazoneId(tinystr::tinystr!(4, "utc"))),
-            zone_variant: Some(ZoneVariant::standard()),
+            bcp47_id: Some(TimeZoneBcp47Id(tinystr::tinystr!(8, "utc"))),
         }
     }
 
-    /// Parse a [`CustomTimeZone`] from a UTF-8 string representing a UTC offset
+    /// The UTC offset.
+    pub const fn offset(&self) -> Option<UtcOffset> {
+        self.offset
+    }
+
+    /// The BCP47 time-zone identifier
+    pub const fn bcp47_id(&self) -> Option<TimeZoneBcp47Id> {
+        self.bcp47_id
+    }
+
+    /// Parse a [`TimeZone`] from a UTF-8 string representing a UTC offset
     /// or an IANA time zone identifier.
     ///
     /// This is a convenience constructor that uses compiled data. For a custom data provider,
     /// use [`UtcOffset`] or [`TimeZoneIdMapper`] directly.
     ///
-    /// To parse from an IXDTF string, use [`CustomZonedDateTime::try_iso_from_str`].
+    /// To parse from an IXDTF string, use [`ZonedDateTime::try_iso_from_str`].
     ///
     /// ✨ *Enabled with the `compiled_data` Cargo feature.*
     ///
     /// # Examples
     ///
     /// ```
-    /// use icu::timezone::CustomTimeZone;
+    /// use icu::timezone::TimeZone;
     /// use icu::timezone::UtcOffset;
     ///
-    /// let tz0: CustomTimeZone = CustomTimeZone::try_from_str("Z")
+    /// let tz0: TimeZone = TimeZone::try_from_str("Z")
     ///     .expect("Failed to parse a time zone");
-    /// let tz1: CustomTimeZone = CustomTimeZone::try_from_str("+02")
+    /// let tz1: TimeZone = TimeZone::try_from_str("+02")
     ///     .expect("Failed to parse a time zone");
-    /// let tz2: CustomTimeZone = CustomTimeZone::try_from_str("-0230")
+    /// let tz2: TimeZone = TimeZone::try_from_str("-0230")
     ///     .expect("Failed to parse a time zone");
-    /// let tz3: CustomTimeZone = CustomTimeZone::try_from_str("+02:30")
+    /// let tz3: TimeZone = TimeZone::try_from_str("+02:30")
     ///     .expect("Failed to parse a time zone");
     ///
-    /// assert_eq!(tz0.offset.map(UtcOffset::offset_seconds), Some(0));
-    /// assert_eq!(tz1.offset.map(UtcOffset::offset_seconds), Some(7200));
-    /// assert_eq!(tz2.offset.map(UtcOffset::offset_seconds), Some(-9000));
-    /// assert_eq!(tz3.offset.map(UtcOffset::offset_seconds), Some(9000));
+    /// assert_eq!(tz0.offset().map(UtcOffset::offset_seconds), Some(0));
+    /// assert_eq!(tz1.offset().map(UtcOffset::offset_seconds), Some(7200));
+    /// assert_eq!(tz2.offset().map(UtcOffset::offset_seconds), Some(-9000));
+    /// assert_eq!(tz3.offset().map(UtcOffset::offset_seconds), Some(9000));
     /// ```
     ///
-    /// [`CustomZonedDateTime::try_iso_from_str`]: crate::CustomZonedDateTime::try_iso_from_str
+    /// [`ZonedDateTime::try_iso_from_str`]: crate::ZonedDateTime::try_iso_from_str
     #[cfg(feature = "compiled_data")]
     #[inline]
     pub fn try_from_str(s: &str) -> Result<Self, UnknownTimeZoneError> {
```

Comment:
Thought: We should support parsing `-0500[America/Chicago]`

Reply:
We should, we already have the code through ixdtf.